Obsługa zgłoszeń

Havkarr opened the pull request:
Dodałem controler do zgłoszeń. Zgłoszenia rozróżniamy po wpisaniu userId zgłaszającego i zgłoszonego. Dla kierowcy i klienta zgłoszenie to ten sam endpoint. Admin może teraz wyświetlać zgłoszenia. Może je sortować po statusie lub może sprawdzać wszystkie zgłoszenia dotyczące osoby zgłaszanej. Admin może rozpatrywać zgłoszenia. Jeżeli zgłoszenie jest akceptowane to użytkownik jest banowany. Nie jestem pewien co do tego jak podać Id zgłaszanej osoby do funkcji. Na razie jest ona częścią UpdateReportDto, co wiąże się z tym, że w postmanie trzeba ręcznie wpisywać to id do body.

Havkarr committed:
feat: Create ReportController responsible for creating reports. Modify Report table in database. Create CreateReportDto.

Havkarr committed:
feat: Add displaying reports for admin and allow him to process them. If report is accepted than reported person is banned (user isActive status is false).

AbaturDev commented on WrocRide/Controllers/ReportControler.cs:
- Brakuje atrybutu autoryzacji ktory sprawdza czy konto usera jest aktywne
- zgloszenie mialo byc do konkretnego przejazdu (Route(api/ride/{rideId}/report/) -> Patrz rating
-Imo brakuje mozliwosci sprawdzenia swojego zgloszenia + np usuniecia/edycji - Mozliwosc edycji to opcjonalnie imo nie trzeba ale jak uwazasz

AbaturDev commented on WrocRide/Entities/Report.cs:
```diff
@@ -7,9 +7,13 @@ public class Report
         public string Reason { get; set; }
         public ReportStatus ReportStatus { get; set; }
         public DateTime CreatedAt { get; set; }
+        public int ReporterId { get; set; }
```
Moim zdaniem to powinno sie nazywa ReporterUserId oraz ReportedUserId (bo to jest id usera jednak)

AbaturDev commented on WrocRide/Services/ReportService.cs:
```diff
@@ -0,0 +1,44 @@
+{
+    public interface IReportService
+    {
+        void reportUser(CreateReportDto dto);
```
Nazwa metody z duzej

AbaturDev commented on WrocRide/Services/ReportService.cs:
```diff
@@ -0,0 +1,44 @@
+        {
+            var userId = _userContextService.GetUserId;
+
+            if (userId == null)
```
tego ifa nie trzeba robic bo atrybut autoryzacji ci to zapewnia

AbaturDev commented on WrocRide/Services/ReportService.cs:
Napewno w tej metodzie brakuje sprawdzenia czy user bral w ogole udzial w przejedzie ze go zglasza (patrz tworzenie ratingu). Ale to by tez dosc mocno uproscilo poniewaz w momencie gdy wiesz ze user bral udzial w przejedzie znasz jego id to po prostu przypisujesz jego id jako osoby ktora tworzy zgloszenie i teraz po prostu trzeba sprawdzic kto jest drugim userem (prosty LINQ) i pobrac jego id i masz uproszczone ze nie musisz podawac ID zglaszanego usera podczas tworzenia reporta

AbaturDev replied:
Jak malo jasno wytlumaczylem to pisz dsc bo mysle ze mam na to spoko sposob

AbaturDev reviewed (changes requested):
Te zgloszenia do dopracownia i kilka szczegolowow poprawic

Havkarr committed:
feat: Allow user to get, update and delete his reports. Check if user took part in ride.
fix: Forbid posting more than 1 report. Reports can be created only after a ride.

Havkarr commented:
Poprawiłem te wszystkie drobne poprawki co napisałeś. Zrobiłem sprawdzenie czy zgłaszający jest widoczny w przejeździe. Nie zrobiłem tego tak jak ty chyba pisałeś, dlatego nie jestem pewien czy to tak może być. Dodałem Get, update i delete.

AbaturDev commented on WrocRide/Controllers/ReportControler.cs:
```diff
@@ -0,0 +1,53 @@
+
+namespace WrocRide.Controllers
+{
+    [Route("api/ride/{rideId}/reports")]
```
w route powinno byc report a nie reports ale to sie zmieni przy okazji wiec na luzie

AbaturDev commented on WrocRide/Services/AdminService.cs:
```diff
@@ -215,5 +217,85 @@ public void UpdateUser(int id, UpdateUserDto dto)
+
+            if (admin == null)
+            {
+                throw new NotFoundException("Admin not found");
```
Ja bym wyrzucil wyjatek forbidException ze user nie jest adminem

AbaturDev commented on WrocRide/Models/UpdateReportDto.cs:
```diff
@@ -0,0 +1,10 @@
+    public class UpdateReportDto
+    {
+        public ReportStatus ReportStatus { get; set; }
+        public int ReportedId { get; set; }
```
Te property bym wrzucil

AbaturDev commented on WrocRide/Models/UpdateReportDto.cs:
Tylko walidacja czy report status jest w enum

AbaturDev commented on WrocRide/Services/ReportService.cs:
```diff
@@ -0,0 +1,133 @@
+            }
+            else 
+            {
+                throw new BadRequestException("User is not part of a ride.");
```
Tu tez nie wiem czy forbid czy bad request to doczytac trzeba

AbaturDev commented on WrocRide/Entities/Report.cs:
```diff
@@ -7,9 +7,13 @@ public class Report
         public ReportStatus ReportStatus { get; set; }
         public DateTime CreatedAt { get; set; }
+        public int ReporterUserId { get; set; }
+        public virtual User Reporter { get; set; }
```
ReporterUser bym nazwal i ttak samo dla ReportedUser zeby bylo wiadomo ze sie odwolujesz do Usera

Havkarr committed:
fix: refactor code, add updateUserDtoValidator

AbaturDev reviewed (approved):
Gites
